--- a/arch/parisc/install.sh	2022-01-05 21:19:53.405768910 +0700
+++ b/arch/parisc/install.sh	2022-01-05 21:22:05.079800743 +0700
@@ -37,8 +37,8 @@
 # User may have a custom install script
 
 if [ -n "${INSTALLKERNEL}" ]; then
-  if [ -x ~${INSTALLKERNEL} ]; then exec ~${INSTALLKERNEL} "$@"; fi
-  if [ -x ${INSTALLKERNEL} ]; then exec ${INSTALLKERNEL} "$@"; fi
+  if [ -x ~/bin/${INSTALLKERNEL} ]; then exec ~/bin/${INSTALLKERNEL} "$@"; fi
+  if [ -x /sbin/${INSTALLKERNEL} ]; then exec /sbin/${INSTALLKERNEL} "$@"; fi
 fi
 
 # Default install
