diff --git a/src/patcher.js b/src/patcher.js
index d1609554029..780740d7423 100644
--- a/src/patcher.js
+++ b/src/patcher.js
@@ -22,7 +22,8 @@ console.log('Hello from Powercord!');
 
 let disableMediaHandling = false;
 try {
-  settings = require(join(__dirname, '../settings/pc-general.json'));
+  const { SETTINGS_FOLDER } = require('./fake_node_modules/powercord/constants');
+  settings = require(join(SETTINGS_FOLDER, 'pc-general.json'));
   disableMediaHandling = settings.disableMediaHandling;
 } catch (e) {}
 
